### PR TITLE
Add TypeDoc references

### LINKED ISSUE #256: Document code specifications with TypeDoc
We should do the documentation for the packages with Typedoc(https://typedoc.org/) which will autogenerate the documentation out of the code documentation.
There are some useful [plugins](https://www.npmjs.com/search?q=keywords:typedocplugin) for typedoc as well, e.g. https://www.npmjs.com/package/typedoc-plugin-markdown (with option for docusaurus docs)

TypeDocs should be generated for
- Lisk Elements packages
- The Lisk Framework
  - Lisk Modules
  - Lisk Plugins

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #256

### CODE REVIEW

File: modules/ROOT/pages/modules/index.adoc [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
-* Handling registerDelegate, voteDelegate, unlockToken, and reportDelegateMisbehavior transaction assets
-* Setting the next delegates set
+| {url_module_base}[Base module]
+| All other modules must extend from the Base module.
+
+| {url_module_auth}[Auth module]
+|Responsible for handling and verifying nonces and for transaction signature validation, including transactions from multi-signature accounts.
+
+| {url_module_fee}[Fee module]
+|Responsible for handling the fee of transactions.
```
Comment (chris529) [inline modules/ROOT/pages/modules/index.adoc]: I think we should reword this heading to the following:
Responsible for handling the transaction fees.

File: modules/ROOT/pages/modules/index.adoc [hunk 2]
```diff
@@ -1,33 +1,54 @@
...
+It allows chains to choose the token used to pay the fee and to define a minimum fee for transactions to be valid.
+
+| {url_module_mci}[Mainchain Interoperability module]
+|Provides basic functionalities for the mainchain to transmit information between interoperable chains in the Lisk ecosystem using cross-chain messages.
+
+| {url_module_pos}[PoS module]
+|Handles all aspects of the generator selection, this includes the registration of accounts as validators, the staking process, and potential reports of misbehavior.
```
Comment (chris529) [inline modules/ROOT/pages/modules/index.adoc]: imho, I think we should have either one of the following:
'Handles all aspects of the generator selection, this includes the registration of accounts for validators, the staking process, and potential reports of misbehavior.'

Or alternatively:

'Handles all aspects of the generator selection, this includes the registration of validator accounts, the staking process, and potential reports of misbehavior.'

File: modules/ROOT/pages/modules/index.adoc [hunk 3]
```diff
@@ -1,33 +1,54 @@
...
+
+| {url_module_sci}[Sidechain Interoperability module]
+|Provides basic functionalities for a sidechain to transmit information between interoperable chains in the Lisk ecosystem using cross-chain messages.
+
+| {url_module_random}[Random module]
+a|Handles the validation of the inputs and computation of outputs for the commit and reveal process of a Lisk blockchain.
+
+| {url_module_reward}[Reward module]
+a|Provides the base reward system for a blockchain developed with the Lisk SDK
```
Comment (chris529) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
a|Provides the base reward system for a blockchain developed with the Lisk SDK.
```

File: modules/ROOT/pages/modules/index.adoc [hunk 4]
```diff
@@ -1,33 +1,54 @@
...
+
+| {url_module_token}[Token module]
+|Handles minting, burning, and transferring tokens int he Lisk ecosystem.
```
Comment (chris529) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Handles minting, burning, and transferring tokens into the Lisk ecosystem.
```

File: modules/ROOT/pages/modules/index.adoc [hunk 5]
```diff
@@ -1,33 +1,54 @@
...
+
+| {url_module_validators}[Validators module]
+|Rresponsible for validating the eligibility of a validator for generating a block and the block signature.
```
Comment (chris529) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Responsible for validating the eligibility of a validator for generating a block and the block signature.
```

File: modules/ROOT/pages/plugins/index.adoc [hunk 6]
```diff
@@ -1,42 +1,36 @@
...
-| xref:{url_plugin_dashboard}[]
+| {url_plugin_dashboard}[Lisk Dashboard plugin]
 |no
 |Provides a user interface which enables developers to conveniently view and communicate with the blockchain application during the development.
 
-| xref:{url_plugin_faucet}[]
+| {url_plugin_faucet}[Lisk Faucet plugin]
 |no
-|Provides a user interface to receive a certain amount of tokens on request.
+|Provides a user interface to distributing testnet tokens from sidechain.
```
Comment (chris529) [inline modules/ROOT/pages/plugins/index.adoc]:
```suggestion
|Provides a user interface for distributing testnet tokens from a sidechain.
```

File: modules/ROOT/pages/references/lisk-commander/index.adoc [hunk 7]
```diff
@@ -74,46 +71,44 @@ Upon successful completion, NPM will add the Lisk Commander executable to your P
...
   passphrase  Commands relating to Lisk passphrases.
 
 COMMANDS
-  autocomplete        display autocomplete installation instructions
-  console             Lisk interactive REPL (Read-eval-print loop), session to run commands.
-  hash-onion          Create hash onions to be used by the forger.
-  help                display help for lisk
-  init                Bootstrap a blockchain application using Lisk SDK.
-  network-identifier
+  autocomplete  display autocomplete installation instructions
```
Comment (chris529) [inline modules/ROOT/pages/references/lisk-commander/index.adoc]:
```suggestion
  autocomplete  Display autocomplete installation instructions.
```

File: modules/ROOT/pages/modules/index.adoc [hunk 8]
```diff
@@ -1,33 +1,54 @@
...
-
-* Snapshotting vote weights
-* Calculating productivity
-* Handling registerDelegate, voteDelegate, unlockToken, and reportDelegateMisbehavior transaction assets
-* Setting the next delegates set
+| {url_module_base}[Base module]
+| All other modules must extend from the Base module.
+
+| {url_module_auth}[Auth module]
+|Responsible for handling and verifying nonces and for transaction signature validation, including transactions from multi-signature accounts.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Responsible for handling and verifying nonces and transaction signature validation, including transactions from multi-signature accounts.
```

File: modules/ROOT/pages/modules/index.adoc (hunk 4 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Handles minting, burning, and transferring tokens in the Lisk ecosystem.
```

File: modules/ROOT/pages/modules/index.adoc (hunk 5 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Responsible for validating the eligibility of a validator for generating a block and the block signature.
```

File: modules/ROOT/pages/plugins/index.adoc [hunk 9]
```diff
@@ -1,42 +1,36 @@
...
 [cols="1,1,2",options="header",stripes="hover"]
 |===
 |Name
 |Included in the Lisk SDK by default?
 |Description
 
-| xref:{url_plugin_dashboard}[]
+| {url_plugin_dashboard}[Lisk Dashboard plugin]
 |no
 |Provides a user interface which enables developers to conveniently view and communicate with the blockchain application during the development.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/plugins/index.adoc]:
```suggestion
|Provides a user interface that enables developers to conveniently view and communicate with the sidechain application during development.
```

File: modules/ROOT/pages/plugins/index.adoc (hunk 6 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/plugins/index.adoc]:
```suggestion
|Provides a user interface for distributing testnet tokens from the sidechain.
```

File: modules/ROOT/pages/references/lisk-commander/index.adoc [hunk 10]
```diff
@@ -74,46 +71,44 @@ Upon successful completion, NPM will add the Lisk Commander executable to your P
...
-  lisk-commander/5.1.10 darwin-x64 node-v16.15.0
+  lisk-commander/6.0.0 darwin-x64 node-v16.18.0
 
 USAGE
   $ lisk [COMMAND]
 
 TOPICS
-  account     Commands relating to Lisk accounts.
+  endpoint    Invokes the provided endpoint.
   generate    Commands relating to Lisk generator.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc]:
```suggestion
  generate    Commands relating to the Lisk generator.
```

File: modules/ROOT/pages/references/lisk-commander/index.adoc (hunk 7 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc]:
```suggestion
  autocomplete  Display autocomplete installation instructions.
```

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 11]
```diff
@@ -135,7 +133,7 @@ The following dependencies need to be installed to run the xref:{url_lisk_sdk}[L
 === Pre-installation
 
 Lisk Elements requires {url_nodejs}[Node.js^] as the underlying engine for code execution.
-This document describes how to install Node.js and xref:{url_npm_glossary}[NPM] for installation via NPM.
+This document describes how to install Node.js and NPM for installation via NPM.
 
 ==== Install node
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:139]: **Line 158**
"As all packages in Lisk Elements are independent **of** each other, they can be installed **separately**."

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 12]
```diff
@@ -57,7 +55,7 @@ Lisk Elements is a collection of JavaScript libraries, each tackling a certain t
 
 | Client
 | {url_npm_lisk_client}[@liskhq/lisk-client^]
-| xref:{url_npm_lisk_client_docs}[lisk-client]
+| {url_npm_lisk_client_docs}[lisk-client^]
 |===
 
 [NOTE]
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:61]: **Line 66**
`apiClient`, `codec`, `cryptography`, `passphrase`, `transactions`, `utils`, `tree`, and `validator`.

File: modules/ROOT/pages/references/lisk-elements/index.adoc (hunk 11 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:139]: **Line 250**
`npm update @liskhq/<package-name>`

File: modules/ROOT/pages/modules/index.adoc (hunk 3 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
a|Provides the base reward system for a blockchain developed with the Lisk SDK.
```

File: modules/ROOT/pages/modules/index.adoc [hunk 13]
```diff
@@ -1,33 +1,54 @@
...
+
+| {url_module_reward}[Reward module]
+a|Provides the base reward system for a blockchain developed with the Lisk SDK
+
+| {url_module_token}[Token module]
+|Handles minting, burning, and transferring tokens int he Lisk ecosystem.
+
+| {url_module_validators}[Validators module]
+|Rresponsible for validating the eligibility of a validator for generating a block and the block signature.
+Furthermore, it maintains information about the registered validators in its module store and provides the generator list.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/modules/index.adoc]:
```suggestion
Furthermore, it maintains information about the registered validators in its module store and provides the validator list.
```

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 14]
```diff
@@ -124,7 +122,7 @@ It is a collection of the following packages:
 
 === Dependencies
 
-The following dependencies need to be installed to run the xref:{url_lisk_sdk}[Lisk SDK]:
+The following dependencies need to be installed to run the Lisk SDK:
 
 [options="header",]
 |===
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:128]: **Line 130**
`|Node.js | v16.18.0 (latest LTS)`

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 15]
```diff
@@ -135,7 +133,7 @@ The following dependencies need to be installed to run the xref:{url_lisk_sdk}[L
 === Pre-installation
 
 Lisk Elements requires {url_nodejs}[Node.js^] as the underlying engine for code execution.
-This document describes how to install Node.js and xref:{url_npm_glossary}[NPM] for installation via NPM.
+This document describes how to install Node.js and NPM for installation via NPM.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc]:
```suggestion
The following section describes how to install dependencies such as Node.js and NPM.
```

**Review by chris529 — COMMENTED** [review]
Looks great! Just a few minor points 👍

File: modules/ROOT/pages/modules/index.adoc (hunk 4 above)
Comment (Tschakki) [reply modules/ROOT/pages/modules/index.adoc]:
```suggestion
|Handles minting, burning, and transferring tokens inside the Lisk ecosystem.
```

File: modules/ROOT/pages/plugins/index.adoc (hunk 6 above)
Comment (Tschakki) [reply modules/ROOT/pages/plugins/index.adoc]:
```suggestion
|Provides a user interface to distributing testnet tokens for a sidechain.
```

File: modules/ROOT/pages/references/lisk-commander/index.adoc (hunk 7 above)
Comment (Tschakki) [reply modules/ROOT/pages/references/lisk-commander/index.adoc]: This is autogenerated

File: modules/ROOT/pages/references/lisk-elements/index.adoc (hunk 11 above)
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:139]: Line 180
Contains an unrendered call-out, maybe we can remove it as we have done the same on other docs as well.

Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc:139]: **Line 231**
`const api = lisk.apiClient.createWSClient('ws://localhost:7887/rpc-ws');`

File: modules/ROOT/pages/references/lisk-commander/index.adoc [hunk 16]
```diff
@@ -22,7 +20,7 @@ Lisk Commander allows communication from the command line with a remote or local
 * Decrypting/encrypting passphrases or messages.
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc:20]: Line 19
`* Bootstrapping a sidechain application.`

File: modules/ROOT/pages/references/lisk-commander/index.adoc [hunk 17]
```diff
@@ -22,7 +20,7 @@ Lisk Commander allows communication from the command line with a remote or local
...
 * Creating new accounts.
 * And much more...
-For further details please see the {url_commander_cli}[command reference^].
+For further details please see the xref:{url_commander_cli}[].
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc:24]:
```suggestion

For further details please see the xref:{url_commander_cli}[].
```

File: modules/ROOT/pages/references/lisk-commander/index.adoc [hunk 18]
```diff
@@ -22,7 +20,7 @@ Lisk Commander allows communication from the command line with a remote or local
...
 
 ---
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc:27]: Line 38
|Node.js | v16.18.0 (latest LTS)

Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-commander/index.adoc:27]: **Line 29 & 47**
Is `8.3` the correct node version?

**Review by TalhaMaliktz — CHANGES_REQUESTED** [review]
Looking good! A few suggestions 😊

File: modules/ROOT/pages/references/lisk-commander/index.adoc (hunk 16 above)
Comment (Tschakki) [reply modules/ROOT/pages/references/lisk-commander/index.adoc:20]: Updated to `sidechain client` to be more precise

File: modules/ROOT/pages/references/lisk-commander/index.adoc (hunk 18 above)
Comment (Tschakki) [reply modules/ROOT/pages/references/lisk-commander/index.adoc:27]: not sure if this is still valid, I have removed the statements for now.

**Comment by Tschakki** [comment]
Thanks, I addressed you feedback so far! But updating the Commander and Elements index is not really part of this PR, we can address it in another issue imho.

File: modules/ROOT/pages/references/lisk-elements/index.adoc [hunk 19]
```diff
@@ -218,7 +217,7 @@ Simply import, (or require) the package and access its functionality according t
 ----
 import lisk from '@liskhq/lisk-client';
 //or
-const lisk = require('@liskhq/lisk-client');
+const lisk = require('@liskhq/lisk-codec');
```
Comment (TalhaMaliktz) [inline modules/ROOT/pages/references/lisk-elements/index.adoc]: I would keep the previous version of this line as in the caption we suggest: "Example with the client package"
```suggestion
const lisk = require('@liskhq/lisk-client');
```

**Review by TalhaMaliktz — APPROVED** [review]
Sure, just a minor change as I think it can be made right away. The rest of it LGTM👍🏼.

File: modules/ROOT/pages/references/lisk-elements/index.adoc (hunk 19 above)
Comment (chris529) [reply modules/ROOT/pages/references/lisk-elements/index.adoc]: 👍

**Review by chris529 — APPROVED** [review]
LGTM! Agree with Talha 👍

**Pull request merged**